### PR TITLE
Support Solidity 0.8.1, 0.8.2 and 0.8.3

### PR DESCRIPTION
**Pull request by blitz-1306**
## Changes
- [x] Bumped dependencies to support Solidity 0.8.1, 0.8.2, 0.8.3 (most important one is solc-typed-ast to v4.0.0).
- [x] Fixed generation of wrapper functions to not contain empty docstrings.
- [x] Tweaked affected test samples.
- [x] Fixed crashes of `merge()` and `sanity()` on sources with `UncheckedBlock`.
- [x] Moved `*.vm.json` and `*.invalid.sol` files to subdirectories to cleanup `samples/` a bit.
- [x] Introduced `if_updated_unsafe.sol` test sample.

## Notes
Still need a few test samples for Solidity 0.8.3 edge cases.

Regards.

### CODE REVIEW

File: test/samples/if_updated_unsafe.sol [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    //// if_assigned {:msg "A1"} x > 0;
+    //// if_updated {:msg "U1"} x > 0;
+    uint8 x = 100;
+
+    /// if_succeeds {:msg "S1"} x == 1;
+    function foo() public {
+        x = 255;
+
+        unchecked {
+            x += 2;
```
Comment (cd1m0) [inline test/samples/if_updated_unsafe.sol:13]: Great job finding this edge case!

**Review by cd1m0 — APPROVED** [review]
LGTM

**Review by cd1m0 — APPROVED** [review]
LGTM
